### DISCUSSION TITLE
provide inside boot file

### DISCUSSION THREAD

**Discussion by botura** [quasarframework/quasar#9978] [General - Components / Directives / etc]
I'm using a third-part component with quasar, so I made a boot file to import and make it available everywhere. But this component also needs a provide statement.

In `App.vue` it looks like this:
```js
  provide: {
    grid: [Edit, Toolbar, Group]
  }
```

But I don't know how to move this code to the boot file. I've tried this without success:
`app.provide('grid', [Edit, Toolbar, Group])`

Is this sintax wrong? How can I debug what is happening?

**Comment by metalsadman** [comment]
@botura what's the error log? if there aren't any, then your library probably uses it in the component level, see provide/inject api https://v3.vuejs.org/api/options-composition.html#provide-inject https://v3.vuejs.org/api/application-api.html#provide (these 2 are different as you can see). if it is, then just add that in your App.vue.

**Reply by botura** [reply]
Thanks @metalsadman ,

There is no error log related to provide, there is a warning complaining about `export 'default' (imported as 'Vue') was not found in 'vue' (possible exports:...`. But I´m not sure if it is the cause, because if I declare the provide at `App.vue` everything works even with the warning. As Syncfusion just released Vue3 compatible components this month, maybe it is something related to their code.

Maybe I just can´t use this provide at a boot file.

I´ve made a repo, if you want and have the time... https://gitlab.com/botura/quasar2-syncfusion

I´ll keep using provide at `App.vue`.

**Reply by rstoenescu** [reply]
`export 'default' (imported as 'Vue') was not found in 'vue' (possible exports:...` --> If this error comes from a file in your /src folder then you have a `import Vue from 'vue'` there (which corresponds to using Vue 2). If this comes from the syncfusion (or whatever else npm package) then it means that the respective package is Vue 2 compatible but NOT compatible with Vue 3.

**Reply by botura** [reply]
It comes from syncfusion package, I´ll open an issue with them.

Thanks for your helping!
